onnxruntime/core/providers/dml/DmlExecutionProvider/src/BucketizedBufferAllocator.h

```diff
@@ -83,16 +83,16 @@ namespace Dml
         std::vector<Bucket> m_pool;
         size_t m_currentAllocationId = 0;
         uint64_t m_currentResourceId = 0;
-        
-        // Unless specifically requested, allocation sizes are not rounded to enable pooling 
-        // until SetDefaultRoundingMode is called.  This should be done at completion of session 
+
+        // Unless specifically requested, allocation sizes are not rounded to enable pooling
+        // until SetDefaultRoundingMode is called.  This should be done at completion of session
         // initialization.
         AllocatorRoundingMode m_defaultRoundingMode = AllocatorRoundingMode::Disabled;
 
         std::shared_ptr<ExecutionContext> m_context;
         std::unique_ptr<DmlSubAllocator> m_subAllocator;
 
-    #if _DEBUG
+    #ifdef _DEBUG
```

Comment:
Technically `#ifndef NDEBUG` (hate the double negative 😅) is standards compliant, as `_DEBUG` is an MSVC specific macro. I know we *currently* don't compile ORT for Linux on WSL with clang, but we have intended to in the past. https://stackoverflow.com/questions/2290509/debug-vs-ndebug

Beware that using `#ifdef SOMEDEBUGFLAG` in general is dangerous because it's legal to `#define SOMEDEBUGFLAG 0`, in which case `#if` works correctly (even if not defined, it will correctly treat it as 0) but `#ifdef` will always treat it as true, even when 0.